eva-accession-release/src/test/java/uk/ac/ebi/eva/accession/release/batch/io/merged/MergedVariantMongoReaderTest.java

```diff
@@ -406,10 +434,82 @@ public void noExceptionIfSsFromDifferentCollectionUpdated() throws Exception {
+        // See scenario here: https://docs.google.com/spreadsheets/d/12QJT4N0-UJGTv3BtVq_gyyrVzweXd5ev2WFlnP-4MW4/edit#rangeid=169520205
+        DbsnpSubmittedVariantEntity ss1 = createSS(ASSEMBLY, TAX1, 1L, 1L, 100L, "C", "A");
+        DbsnpSubmittedVariantEntity ss2 = createSS(ASSEMBLY, TAX2, 2L, 2L, 101L, "A", "T");
+        DbsnpSubmittedVariantEntity ss3 = createSS(ASSEMBLY, TAX2, 3L, 3L, 102L, "T", "G");
+        DbsnpClusteredVariantEntity rs1 = createRS(ss1, null);
+        DbsnpClusteredVariantEntity rs2 = createRS(ss2, null);
+        DbsnpClusteredVariantEntity rs3 = createRS(ss3, TAX1);
+        this.mongoTemplate.insert(Stream.of(ss1, ss2, ss3).map(ss -> {
+            DbsnpSubmittedVariantOperationEntity dbsnpSvoeObj = new DbsnpSubmittedVariantOperationEntity();
+            dbsnpSvoeObj.fill(EventType.UPDATED, ss.getAccession(),
+                              "Original RS was merged into another RS",
+                              Arrays.asList(new DbsnpSubmittedVariantInactiveEntity(ss)));
+            return dbsnpSvoeObj;
+        }).collect(Collectors.toList()), DbsnpSubmittedVariantOperationEntity.class);
+        this.mongoTemplate.insert(Stream.of(rs1, rs2, rs3).map(rs -> {
+            DbsnpClusteredVariantOperationEntity dbsnpCvoeObj = new DbsnpClusteredVariantOperationEntity();
+            dbsnpCvoeObj.fill( EventType.MERGED, rs.getAccession(), rs.getAccession() + 10,
+                               "Original RS was merged into another RS",
+                               Arrays.asList(new DbsnpClusteredVariantInactiveEntity(rs)));
+            return dbsnpCvoeObj;
+        }).collect(Collectors.toList()), DbsnpClusteredVariantOperationEntity.class);
+        // If we don't insert target CVEs of the merges, the merge reader will fail
+        // because such merged RS will be categorized as merged-then-deprecated
+        this.mongoTemplate.insert(Stream.of(rs1, rs2, rs3).map(rs -> new ClusteredVariantEntity(
+                                          rs.getAccession() + 10, rs.getHashedMessage(), rs.getModel()))
+                                        .collect(Collectors.toList()), DbsnpClusteredVariantEntity.class);
+
+        // Test records returned when the mongo reader is given TAX1
+        MergedVariantMongoReader reader  = new MergedVariantMongoReader(ASSEMBLY, TAX1, mongoClient, TEST_DB,
+                                                                        CHUNK_SIZE, new DbsnpCollectionNames());
+        reader.open(new ExecutionContext());
+        List<Variant> mergedVariants = new ArrayList<>(readIntoMap(reader).values());
+        assertEquals(1, mergedVariants.size());
+        assertEquals("rs1", mergedVariants.get(0).getMainId());
+
+        // Test records returned when the mongo reader is given TAX2
+        // Note rs3 is returned even though in Mongo it was originally issued in TAX1
+        // This is because there is a SS record in TAX2 that was declustered from rs3
```

Comment:
```suggestion
        // This is because there is a SS record in TAX2 that was clustered under rs3 before merge
```
or something like that - I think this is what you mean

Reply:
addressed